File: 0090-subsets-ii/0090-subsets-ii.js
/**
 * @param {number[]} nums
 * @return {number[][]}
 */
var solve=function(nums,ans,curr,i){
    if(i==nums.length){
        ans.push([...curr])
        return;
    }
    curr.push(nums[i]);
    solve(nums,ans,curr,i+1);
    curr.pop();
    while(i+1<nums.length && nums[i]==nums[i+1]){
        i++;
    }
    solve(nums,ans,curr,i+1);
}
var subsetsWithDup = function(nums) {
    nums.sort((a,b)=>a-b); 
    let ans=[];
    let curr=[];
    solve(nums,ans,curr,0);
   return ans;
};

File: 0090-subsets-ii/0090-subsets-ii.cpp
class Solution {
public:
    void solve(vector<int>&nums,int index,vector<int>&curr,vector<vector<int>>&ans){
        if(index==nums.size()){
            ans.push_back(curr);
            return;
        }
        curr.push_back(nums[index]);
        solve(nums,index+1,curr,ans);
        curr.pop_back();
        while(index+1<nums.size() && nums[index]==nums[index+1]){
            index++;
        }
        solve(nums,index+1,curr,ans);

    }
    vector<vector<int>> subsetsWithDup(vector<int>& nums) {
        sort(nums.begin(), nums.end());
        vector<int>curr;
        vector<vector<int>>ans;
        solve(nums,0,curr,ans);
        return ans;

    }
};